File: VladislavShalnev/Term 2/Block 4/Task 8 (Bash)/Bash/Commands/EchoCommand.cs
using Bash.Utils;

namespace Bash.Commands;

internal class EchoCommand : ICommand
{
	public string Trigger => "echo";

	private IIOManager _io;

	public EchoCommand(IIOManager io) =>
		_io = io;
	
	public string[]? Run(string[] args)
	{
		_io.Write(string.Join(' ', args.Select(arg => arg.Replace("\"", ""))));

		return null;
	}
}

File: Khelmianov-Artem/Term2/Block4/Bash/src/main/kotlin/commands/EchoCommand.kt
package commands

import bash.Environment
import java.io.InputStream
import java.io.OutputStream

class EchoCommand(args: List<String>) : ACommand("echo", args) {
    override fun run(input: InputStream, output: OutputStream, error: OutputStream, env: Environment): Int {
        output.write(args.joinToString(" ", postfix = System.lineSeparator()).toByteArray())
        return 0
    }
}

